### PR TITLE
Complete Aardvark field method refactor

### PR DESCRIPTION
**Pull request by ehanson8**
### Purpose and background context
Refactors remaining `Aardvark` field methods to align with the conventions developed since this class was initially created.

### How can a reviewer manually see the effects of these changes?
Run the following command to see that the Aardvark class transform still transforms a source file:

```
pipenv run transform -o output/gismit.json -i  tests/fixtures/aardvark/aardvark_record_all_fields.jsonl -s gismit
```

### Includes new or updated dependencies?
NO

### Changes expectations for external applications?
NO

### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/TIMX-285

### Developer
- [x] All new ENV is documented in README
- [x] All new ENV has been added to staging and production environments
- [x] All related Jira tickets are linked in commit message(s)
- [x] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed and verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: transmogrifier/sources/json/aardvark.py [hunk 1]
```diff
@@ -368,11 +369,14 @@ def get_publishers(source_record: dict) -> list[timdex.Publisher]:
...
 
-    @staticmethod
-    def get_rights(source: str, source_record: dict) -> list[timdex.Rights]:
-        """Get values from source record for TIMDEX rights field."""
+    @classmethod
+    def get_provider(cls, source_record: dict) -> str | None:
+        return source_record.get("schema_provider_s") or None
+
+    @classmethod
+    def get_rights(cls, source_record: dict, source: str) -> list[timdex.Rights] | None:
```
Comment (ehanson8) [inline transmogrifier/sources/json/aardvark.py:379]: Perhaps this should be broken into submethods but I wanted to get your thoughts to see if it's worth it

File: transmogrifier/sources/json/aardvark.py [hunk 2]
```diff
@@ -55,10 +55,10 @@ def get_source_link(
...
                     overrides
         """
-        links = cls.get_links(source_record, source_record_id)
-        url_links = [link for link in links if link.kind == "Website"]
-        if len(url_links) == 1:
-            return url_links[0].url
+        if links := cls.get_links(source_record):
+            url_links = [link for link in links if link.kind == "Website"]
+            if len(url_links) == 1:
+                return url_links[0].url
```
Comment (ghukill) [inline transmogrifier/sources/json/aardvark.py:61]: This is a good reminder on how fuzzy things get over time... I don't think we'd ever expect to have multiple `kind=Website` coming back from `gismit` or `gisogm` records?  

If we did, then we might want another scenario handler for when `len() > 1`, but if not, I think it's good as-is.

**UPDATE**: keeping comment, but after looking more closely at `get_links()`, I think we're okay, given that `kind=Website` is only set for a single possible key in the `dct_references_s` JSON/dictionary values.

File: transmogrifier/sources/json/aardvark.py (hunk 1 above)
Comment (ghukill) [reply transmogrifier/sources/json/aardvark.py:379]: I hate to say it, but I think I'd agree.  I could see methods like:
- `get_access_rights()`
- `get_license_rights()`
- `get_rights_and_rights_holders()`
- `get_gismit_rights()` + `get_gisogm_rights()` (and perhaps these triggered by what source logic)

An intersting one for sure though, it's like a big collapsing of multiple fields into one.

**Review by ghukill — APPROVED** [review]
I think it's looking great!  Left a couple of comments, and agree with your suggestion about breaking `get_rights()` into sub-methods.  But feel comfortable approving with any follow-up changes on these fronts.

File: transmogrifier/sources/json/aardvark.py (hunk 1 above)
Comment (ehanson8) [reply transmogrifier/sources/json/aardvark.py:379]: Deep down, I knew it had to be done 🙃

**Comment by ehanson8** [comment]
Prompted a little more refactoring than expected so re-requesting review

File: transmogrifier/sources/json/aardvark.py [hunk 3]
```diff
@@ -368,65 +369,75 @@ def get_publishers(source_record: dict) -> list[timdex.Publisher]:
...
+        rights: list[timdex.Rights] = []
         kind_access_to_files = "Access to files"
+        rights.extend(cls._get_access_rights(source_record))
+        rights.extend(cls._get_license_rights(source_record))
+        rights.extend(cls._get_rights_and_rights_holders(source_record))
+        if source == "gisogm":
+            rights.extend(cls._get_gisogm_rights(kind_access_to_files))
+        elif source == "gismit":
+            rights.extend(cls._get_gismit_rights(source_record, kind_access_to_files))
+        return rights or None
```
Comment (ghukill) [inline transmogrifier/sources/json/aardvark.py:389]: 🔥 Looks great!  The sub-methods are clear, this higher level method is clear, easy to follow.

**Review by ghukill — APPROVED** [review]

File: transmogrifier/sources/json/aardvark.py [hunk 4]
```diff
@@ -55,10 +55,10 @@ def get_source_link(
                 - not used by default implementation, but could be useful for subclass
                     overrides
         """
-        links = cls.get_links(source_record, source_record_id)
-        url_links = [link for link in links if link.kind == "Website"]
-        if len(url_links) == 1:
-            return url_links[0].url
+        if links := cls.get_links(source_record):
+            url_links = [link for link in links if link.kind == "Website"]
```
Comment (jonavellecuerdo) [inline transmogrifier/sources/json/aardvark.py:59]: [Non-blocking] Could be updated to:

```python
if (links := cls.get_links(source_record)) and (url_links := [link for link in links if link.kind == "Website"]
):
   return url_links[0].url
```

**Review by jonavellecuerdo — APPROVED** [review]
Looks good to me! I had one non-blocking suggested syntax change, but feel free to ignore if you find that it is less readable. 🤓

File: transmogrifier/sources/json/aardvark.py (hunk 4 above)
Comment (ehanson8) [reply transmogrifier/sources/json/aardvark.py:59]: Good suggestion, thanks!

**Pull request merged**